### PR TITLE
rustdoc: don't panic when the cross-re-export handler sees a proc-macro

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
When i moved the macro cross-re-export inlining code into `clean::inline`, i thought that if a macro had a `Def` that said it was a bang macro, it wouldn't be a proc macro. I thought wrong. Turns out, the `quote!()` in `libproc_macro` is actually a proc-macro, and when the `quote!()` macro is re-exported, this proc-macro is accessed in its place. This causes any `proc_macro::*` glob re-export to pull in this proc-macro, causing the assertion i added to fire, leading to an ICE. This replaces that with an Option that ignores proc-macros for the time being.

Fixes https://github.com/rust-lang/rust/issues/52129

### CODE REVIEW

File: src/librustdoc/clean/inline.rs [hunk 1]
```diff
@@ -104,7 +104,11 @@ pub fn try_inline(cx: &DocContext, def: Def, name: ast::Name, visited: &mut FxHa
         // separately
         Def::Macro(did, MacroKind::Bang) => {
             record_extern_fqn(cx, did, clean::TypeKind::Macro);
-            clean::MacroItem(build_macro(cx, did, name))
+            if let Some(mac) = build_macro(cx, did, name) {
```
Comment (xfix) [inline src/librustdoc/clean/inline.rs]: Out of curiosity, is there a reason why it cannot be `clean::MacroItem(build_macro(cx, did, name)?);`

Comment (QuietMisdreavus) [reply src/librustdoc/clean/inline.rs]: Because i never use the question mark operator, so i never thought about it! 😅

**Pull request merged**